.github/workflows/e2e-windows.yaml

```diff
@@ -97,3 +123,62 @@ jobs:
           make clean
           cd deps/finch-core && make clean
           exit 0 # Cleanup may set the exit code e.g. if a file doesn't exist; just ignore
+  upload:
```

Comment:
Can this job be moved to CI workflow and upload all the reports from linux, macOS, and Windows?

Reply:
Discussed offline, this is fine for now even if it is slightly duplicative. Can we just update the job name to be more explicit?